pkg/oauth/key_exists_middleware.go

```diff
@@ -19,12 +20,13 @@ var (
 
 // KeyExistsMiddleware checks the integrity of the provided OAuth headers
 type KeyExistsMiddleware struct {
-	manager *Manager
+	manager       *Manager
+	oAuthServerID bson.ObjectId
```

Comment:
Passing the `api.Spec` pointer is a bit more flexible and probably will allow us to check for other things in the future